### ISSUE TITLE
DCS World Crashes on Launch with Varjo-Foveated Rendering installed

### ISSUE THREAD

**Issue by johnobrien** [mbucchia/Varjo-Foveated#3]
On 8/14/2023, everything worked fine. On 8/26/2023, DCS World crashes on launch w/ Varjo Foveated installed, but does not if I uninstall Varjo-Foveated. 

I have attempted rolling back an nVidia Driver update that happened in the interim, which has not resolved the issue. My log file for Varjo-Foveated is as follows:

2023-08-26 19:07:41 -0400: XR_APILAYER_MBUCCHIA_varjo_foveated layer (v0.2.0) is active
2023-08-26 19:07:41 -0400: Using layer: XR_APILAYER_MBUCCHIA_varjo_foveated
2023-08-26 19:07:41 -0400: Requested extension: XR_KHR_D3D11_enable
2023-08-26 19:07:41 -0400: Requested extension: XR_KHR_visibility_mask
2023-08-26 19:07:41 -0400: Requested extension: XR_VARJO_quad_views
2023-08-26 19:07:42 -0400: Requesting extension: XR_VARJO_foveated_rendering
2023-08-26 19:07:42 -0400: Application: DCS World
2023-08-26 19:07:42 -0400: Using OpenXR runtime: Varjo OpenXR Runtime 3.10.1
2023-08-26 19:07:42 -0400: Using OpenXR system: AERO
2023-08-26 19:07:42 -0400: supportsFoveatedRendering = 1
2023-08-26 19:07:42 -0400: Trying to locate configuration file at 'C:\Users\johno\AppData\Local\Varjo-Foveated\settings.cfg'...
2023-08-26 19:07:42 -0400: Not found
2023-08-26 19:07:42 -0400: Trying to locate configuration file at 'C:\Program Files\OpenXR-Varjo-Foveated\settings.cfg'...
2023-08-26 19:07:42 -0400:   Found option 'peripheral_multiplier=1'
2023-08-26 19:07:42 -0400:   Found option 'focus_multiplier=1'
2023-08-26 19:07:42 -0400:   Found option 'horizontal_focus_scale=1'
2023-08-26 19:07:42 -0400:   Found option 'vertical_focus_scale=1'
2023-08-26 19:07:42 -0400:   Found option 'turbo_mode=1'
2023-08-26 19:07:42 -0400:   Found option 'no_eye_tracking=0'
2023-08-26 19:07:42 -0400: Peripheral resolution: 1092x936 (multiplier: 1.000)
2023-08-26 19:07:42 -0400: Focus resolution 922x922 (multiplier: 1.000/1.000)
2023-08-26 19:07:42 -0400: Peripheral resolution: 1092x936 (multiplier: 1.000)
2023-08-26 19:07:42 -0400: Focus resolution 922x922 (multiplier: 1.000/1.000)
2023-08-26 19:07:44 -0400: Creating swapchain with resolution: 1092x936
2023-08-26 19:07:44 -0400: Creating swapchain with resolution: 1092x936
2023-08-26 19:07:44 -0400: Creating swapchain with resolution: 922x922
2023-08-26 19:07:44 -0400: Creating swapchain with resolution: 922x922
2023-08-26 19:07:46 -0400: Application is using Quad Views for this session.
2023-08-26 19:07:50 -0400: xrEndFrame failed with XR_ERROR_RUNTIME_FAILURE

**Comment by mbucchia** [comment]
Do you have other OpenXR API layers installed? Can you check with https://github.com/maluoi/openxr-explorer/releases/download/v1.4/openxr-explorer-win-x64.zip and post a screenshot of that?

**Comment by johnobrien** [comment]
Attached please find the requested screenshot. Please let me know if I can help provide any other information.
![Screenshot 2023-08-26 204849](https://github.com/mbucchia/Varjo-Foveated/assets/446719/41a9cc27-ac1d-49af-b1cf-921c8fcaac06)

**Comment by mbucchia** [comment]
Thanks. Not really seeing anything wrong, and nobody had reported such error before. Are you on some sort of private bets of DCS?

**Comment by johnobrien** [comment]
No. I am running the non-Steam Open Beta, multithreaded binary, version
2.8.8.43704.

The whole thing is baffling; other than the graphics driver which I tried
rolling back (which didn't fix the issue), I can't think of a single thing
I've changed on the system between 8/13 and 8/26.

**Comment by mbucchia** [comment]
There is an extra tracing we can do, but it's a little complicated. 

1) start a PowerShell prompt as Administrator
2) `cd $env:localappdata\temp`
3) start recording: `wpr -start $env:programfiles\openxr-varjo-foveated\tracing.wprp -filemode`
4) run DCS (your normal way), repro crash
5) stop recording: `wpr -stop varjo-foveated.etl`
6) ZIP up the file `%localappdata%\temp\varjo-foveated.etl` and send it to me

**Comment by johnobrien** [comment]
Attached please find the trace file requested. Please let me know if I can provide any other information.
[varjo-foveated.zip](https://github.com/mbucchia/Varjo-Foveated/files/12447160/varjo-foveated.zip)

**Comment by johnobrien** [comment]
And now it is working.... I don't know why is wasn't or why it is now... Bizzare.

**Comment by mbucchia** [comment]
Haven't looked at the trace yet. Can you attach a working log?

Did you happen to change the motion smoothing settings, or any other settings in Varjo Base?

The only crash that comes close to this that was reported to me was about motion smoothing and low resolution (and it is a Varjo issue).

**Comment by johnobrien** [comment]
Yes in fact I did. I had Motion Smoothing Always enabled set in Varjo Base,
and I changed it to Enable if App supports it. I was toggling it for some
other game, wow so that was it?

Attached please find the log from varjo-foveated after it launched
successfully.

**Comment by mbucchia** [comment]
I don't know for sure,  but I've heard of a crash in Varjo software when the resolution is low. Based on your initial log, your PPD setting is in the lower range(?)

This is a bug we should report to Varjo. I will look at your trace tomorrow and see if there is anything I can use for a bug report.

Thanks!

**Comment by mbucchia** [comment]
Reopening for safe keeping and so I dont forget to follow up.

**Comment by mbucchia** [comment]
Looking at the trace - the Varjo OpenXR runtime is returning quite bogus data. I suspect their motion smoothing algorithm is getting confused by Varjo-Foveated's Turbo Mode.

**Comment by johnobrien** [comment]
Interesting. I didn't have Turbo Mode turned on (at least not
intentionally). Does the trace say I did?

**Comment by mbucchia** [comment]
It's on by default. Your log shows it on

**Comment by johnobrien** [comment]
ah. Well there you go.
